cogs/source/source.py

```diff
@@ -8,34 +8,60 @@ async def postSourceFunction(ctx, imageURL):
     try:
         tracemoe = tracemoepy.tracemoe.TraceMoe()
         result = tracemoe.search(imageURL.strip("<>"), is_url=True)
-        titleEnglish = result.docs[0].title_english
-        anilistID = f"{result.docs[0].anilist_id}"
-        episode = f"{result.docs[0].episode}"
-        similarity = float(result.docs[0].similarity)
+        titleEnglish = result.docs[0].title_english or "No Title Found"
+        anilistID = f"{result.docs[0].anilist_id}" or "No anilistID Found"
+        episode = f"{result.docs[0].episode}" or "No Episode Found"
+        similarity = float(result.docs[0].similarity) or 0
+
         URL = "https://anilist.co/anime/" + anilistID
 
```

Comment:
To help with consolidating, consider:
```suggestion
        if result.docs[0].title_english:
            URL = "https://anilist.co/anime/" + anilistID
        else:
            URL = ""
```
or
```suggestion
        URL = ""
        if result.docs[0].title_english:
            URL = "https://anilist.co/anime/" + anilistID
```

Reply:
Pulled the message into a helper function that should help avoid the duplicated code